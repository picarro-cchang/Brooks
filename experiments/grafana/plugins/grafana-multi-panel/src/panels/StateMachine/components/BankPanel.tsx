import React, {Component, PureComponent} from 'react';

interface BankPanelOptions {
    bank: number,
    uistatus: {
        bank?: {[bankNum:string]: string}
        clean?: {[bankNum:string]: string}
        channel?: {[bankNum:string]: {[channelNum:string]: string}}
    }
    ws_sender: (o: object)=>void;
}

class BankPanel extends PureComponent<BankPanelOptions> {
    bankStyleOpt = {
        READY:{color:"#fff", backgroundColor:"#888"},
        ACTIVE:{color:"#fff", backgroundColor:"#080"},
        CLEAN:{color:"#fff", backgroundColor:"#4BBEE3"},
        REFERENCE:{color:"#440", backgroundColor:"#febb00"},
        DISABLED:{ display: "none" }}; //hope this will work to not display any inactive banks
    cleanClassNameOpt = {
        DISABLED:"btn-inverse disabled", READY: "btn-light", ACTIVE:"btn-primary", CLEAN:"btn-secondary"
    };
    channelClassNameOpt = {
        DISABLED:"btn-inverse disabled", READY: "btn-light", AVAILABLE:"btn-light", ACTIVE:"btn-primary", CLEAN:"btn-primary", REFERENCE:"btn-warning"
    };

    render() {
        let bankStyle = {};
        let cleanClassNames = "";
        let cleanDisabled = true;
        let getChannelClassNames = (_:number) => "";
        let getChannelDisabled = (_:number) => true;
        let test = {}

        if ("bank" in (this.props.uistatus as any)) {
            console.log("In bankPanel.render", this.props.uistatus);
            const bankStatus: string = (this.props.uistatus.bank as any)[this.props.bank];
            const channelStatus: {[key:number]: string} = (this.props.uistatus.channel as any)[this.props.bank];
            const cleanStatus: string = (this.props.uistatus.clean as any)[this.props.bank];
            bankStyle = (this.bankStyleOpt as any)[bankStatus];
            cleanClassNames = (this.cleanClassNameOpt as any)[cleanStatus];
            cleanDisabled = cleanStatus !== "READY";
            getChannelClassNames = (chan) => (this.channelClassNameOpt as any)[(channelStatus as any)[chan]];
            getChannelDisabled = (chan) => (channelStatus as any)[chan] !== "READY";
            test = channelStatus;
        }
       // console.log("hello", channelStatus);

        let channelButtons = [];
        for (let i=1; i<=8; i++) {
            channelButtons.push(
                (getChannelDisabled(i)) ? (
                    <button
                        key={i}
                        className={"btn btn-large " + getChannelClassNames(i)}
                        style={{ height: 90, width: 90,  color: 'black', position: "relative"}}>
                        <p style={{ position: "absolute", top: 20, left: "-0.37em", bottom: 0, right: 0}}>{"Ch " + i}</p>

                        <p style={{fontSize: 10, marginTop: 30}}>Status: {test[i]} </p>


                    </button>
                ) : (
                    <button
                        onClick={e => this.props.ws_sender({element: "channel", bank: this.props.bank, channel: i})}
                        disabled={getChannelDisabled(i)}
                        key={i}
                        className={"btn btn-large " + getChannelClassNames(i)}
                        style={{ height: 90, width: 90,  color: 'black',  position: "relative"}}>
                        <p style={{ position: "absolute", top: 20, left: "-0.37em", bottom: 0, right: 0}}>{"Ch " + i}</p>

                        <p style={{ fontSize: 10, marginTop: 30 }}>Status: {test[i]}</p>
                    </button>
                )
            );
        }

        const cleanButton = (cleanDisabled) ? (
            <div
                style={{width:"100%", border:"3px solid #CCC", color: "black"}}
                className={"btn btn-large " + cleanClassNames}>
                {"Clean"}
            </div>
        ) : (
            <button
                style={{width:"100%", border:"3px solid #CCC", color: 'black'}}
                onClick={e => this.props.ws_sender({element: "clean", bank: this.props.bank})}
                className={"btn btn-large " + cleanClassNames}>
                {"Clean"}
            </button>
        );

        return (
            <div style={{padding: "10px"}}>
                <div className="panel-bank" style={{...{border: "3px solid #111", borderRadius: "10px", padding:"10px", float: "left"},...bankStyle}}>
                    <div style={{width: "100%"}}>
                        <h2>Bank {this.props.bank}</h2>
                        <div className="grid-bank" style={{display: "grid", gridGap: 10, gridTemplateColumns: "1fr 1fr", paddingBottom: 10}}>
                            {channelButtons}
                        </div>
                        {cleanButton}
                    </div>
                </div>
            </div>
        );
    }
}
export default BankPanel;
