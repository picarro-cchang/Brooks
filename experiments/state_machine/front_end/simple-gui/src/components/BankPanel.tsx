import React, {Component, PureComponent} from 'react';

interface BankPanelOptions {
    bank: number,
    uistatus: {
        bank?: {[bankNum:string]: string}
        clean?: {[bankNum:string]: string}
        channel?: {[bankNum:string]: {[channelNum:string]: string}}
    }
}

class BankPanel extends PureComponent<BankPanelOptions> {
    bankStyleOpt = {
        READY:{color:"#fff", backgroundColor:"#888"}, 
        ACTIVE:{color:"#fff", backgroundColor:"#080"}, 
        CLEAN:{color:"#fff", backgroundColor:"#008"}, 
        REFERENCE:{color:"#440", backgroundColor:"#CC0"}};
    cleanClassNameOpt = {
        DISABLED:"btn-light disabled", READY: "btn-light", ACTIVE:"btn-primary", CLEAN:"btn-primary"
    };
    channelClassNameOpt = {
        DISABLED:"btn-light disabled", READY: "btn-light", ACTIVE:"btn-success", CLEAN:"btn-primary", REFERENCE:"btn-light"
    };

    render() {
        let bankStyle = {};
        let cleanClassNames = "";
        let cleanDisabled = true;
        let getChannelClassNames = (_:number) => "";
        let getChannelDisabled = (_:number) => true;

        if ("bank" in (this.props.uistatus as any)) {
            const bankStatus: string = (this.props.uistatus.bank as any)[this.props.bank];
            const channelStatus: {[key:number]: string} = (this.props.uistatus.channel as any)[this.props.bank];
            const cleanStatus: string = (this.props.uistatus.clean as any)[this.props.bank];
            bankStyle = (this.bankStyleOpt as any)[bankStatus];
            cleanClassNames = (this.cleanClassNameOpt as any)[cleanStatus];
            cleanDisabled = cleanStatus === "DISABLED";
            getChannelClassNames = (chan) => (this.channelClassNameOpt as any)[(channelStatus as any)[chan]];
            getChannelDisabled = (chan) => (channelStatus as any)[chan] === "DISABLED";
        }
        let channelButtons = [];
        for (let i=1; i<=8; i++) {
            channelButtons.push(
                <button
                    disabled={getChannelDisabled(i)}
                    className={"btn btn-lg btn-bank " + getChannelClassNames(i)}>
                    Ch {i}
                </button>
            );
        }

        return (

            <div style={{padding: "10px"}}>
                <div className="panel-bank" style={{...{border: "3px solid #111", borderRadius: "10px", padding:"10px"},...bankStyle}}>
                    <div style={{width: "100%"}}>
                        <h2>Bank {this.props.bank}</h2>
                        <div className="grid-bank">
                            {channelButtons}
                        </div>
                        <button style={{width:"100%", border:"3px solid #CCC"}}
                            disabled={cleanDisabled}
                            className={"btn btn-lg " + cleanClassNames}>
                            Clean
                        </button>
                    </div>
                </div>
            </div>
        );
    }
}
export default BankPanel;