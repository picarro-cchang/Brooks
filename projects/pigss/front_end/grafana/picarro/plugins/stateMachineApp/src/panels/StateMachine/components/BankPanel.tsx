import React, { PureComponent } from "react";
import "./bankpanel.css";

export interface BankPanelOptions {
  bank: number;
  uistatus: {
    bank?: { [bankNum: string]: string };
    clean?: { [bankNum: string]: string };
    channel?: { [bankNum: string]: { [channelNum: string]: string } };
  };
  ws_sender: (o: object) => void;
  plan: {
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
  };
}

class BankPanel extends PureComponent<BankPanelOptions> {
  bankStyleOpt = {
    READY: { color: "#424242", backgroundColor: "#e4e4e4" },
    ACTIVE: { color: "#fff", backgroundColor: "#56a64b" },
    CLEAN: { color: "#fff", backgroundColor: "#4BBEE3" },
    REFERENCE: { color: "#440", backgroundColor: "#e0b400" }
  };
  cleanClassNameOpt = {
    DISABLED: "btn-inverse-2 disabled",
    READY: "btn-ready",
    ACTIVE: "btn-light",
    CLEAN: "btn-secondary btn-clean-act"
  };
  channelClassNameOpt = {
    DISABLED: "btn-inverse-2 disabled",
    READY: "btn-ready",
    AVAILABLE: "btn-ready",
    ACTIVE: "btn-green",
    CLEAN: "btn-light",
    REFERENCE: "btn-warning"
  };

  render() {
    let bankStyle = {};
    let cleanClassNames = "";
    let cleanDisabled = true;
    let getChannelClassNames = (_: number) => "";
    let getChannelDisabled = (_: number) => true;
    let test = {};

    if ("bank" in (this.props.uistatus as any)) {
      const bankStatus: string = this.props.uistatus.bank && (this.props.uistatus.bank as any)[
        this.props.bank
      ];
      const channelStatus: { [key: number]: string } = this.props.uistatus.channel && (this.props.uistatus
        .channel as any)[this.props.bank];
      const cleanStatus: string = this.props.uistatus.clean && (this.props.uistatus.clean as any)[
        this.props.bank
      ];
      bankStyle = (this.bankStyleOpt as any)[bankStatus];
      cleanClassNames = (this.cleanClassNameOpt as any)[cleanStatus];
      cleanDisabled = cleanStatus !== "READY";
      getChannelClassNames = chan => {
        return this.channelClassNameOpt !== undefined && 
                channelStatus !== undefined && 
                (this.channelClassNameOpt as any)[(channelStatus as any)[chan]];
      }
      getChannelDisabled = chan => { return channelStatus !== undefined && (channelStatus as any)[chan] !== "READY" };
      test = channelStatus;
    }

    const channelButtons = [];
    for (let i = 1; i <= 8; i++) {
      channelButtons.push(
        getChannelDisabled(i) ? (
          <button
            key={i}
            id={"channel-" + i}
            className={"btn btn-large bank-btn " + getChannelClassNames(i)}
            style={{ color: "black" }}
          >
            <p className="chn-label">
              <u className={"chn-name-" + i}>
                {this.props.plan.bank_names[this.props.bank].channels[i]}
              </u>
            </p>
            <p id={"chn-status-" + i} className={"chn-status"}>
              {" "}
              {test && test[i]}{" "}
            </p>
          </button>
        ) : (
          <button
            onClick={e => {
              this.props.ws_sender({
                element: "channel",
                bank: this.props.bank,
                channel: i
              });
            }}
            id={"channel-" + i}
            disabled={getChannelDisabled(i)}
            key={i}
            className={"btn btn-large bank-btn " + getChannelClassNames(i)}
          >
            <p className="chn-label">
              <u className={"chn-name-" + i}>
                {this.props.plan.bank_names[this.props.bank].channels[i]}{" "}
              </u>
            </p>

            <p id={"chn-status-" + i} className={"chn-status"}>
              {" "}
              {test && test[i]}
            </p>
          </button>
        )
      );
    }

    const cleanButton = cleanDisabled ? (
      <div className={"btn btn-large btn-clean " + cleanClassNames}>
        {"Clean"}
      </div>
    ) : (
      <button
        id="clean"
        onClick={e =>
          this.props.ws_sender({ element: "clean", bank: this.props.bank })
        }
        className={"btn btn-large btn-clean " + cleanClassNames}
      >
        {"Clean"}
      </button>
    );
    const value: string = this.props.plan.bank_names[this.props.bank].name;
    return (
      <div>
        <div className="panel-bank" style={bankStyle}>
          <div style={{ width: "100%" }}>
            <h2 style={{ color: "#424242", marginBottom: "10px" }}>{value}</h2>
            <div className="grid-bank">{channelButtons}</div>
            {cleanButton}
          </div>
        </div>
      </div>
    );
  }
}
export default BankPanel;
