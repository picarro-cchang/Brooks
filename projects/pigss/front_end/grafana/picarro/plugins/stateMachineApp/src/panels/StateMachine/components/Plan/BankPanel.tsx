import React, { PureComponent } from "react";
import { BankPanelPlanOptions, Plan } from "../types";
import "./../bankpanel.css";

interface State {
  plan: Plan;
}

class BankPanel extends PureComponent<BankPanelPlanOptions, State> {
  constructor(props){
    super(props)
    this.state = {
      plan: this.props.plan
    }
  }
  
  bankStyleOpt = {
    READY: { color: "#424242", backgroundColor: "#e4e4e4" },
    ACTIVE: { color: "#fff", backgroundColor: "#56a64b" },
    CLEAN: { color: "#fff", backgroundColor: "#4BBEE3" },
    REFERENCE: { color: "#440", backgroundColor: "#e0b400" }
  };
  cleanClassNameOpt = {
    DISABLED: "btn-inverse-2 disabled",
    READY: "btn-ready",
    ACTIVE: "btn-light",
    CLEAN: "btn-secondary btn-clean-act"
  };
  channelClassNameOpt = {
    DISABLED: "btn-inverse-2 disabled",
    READY: "btn-ready",
    AVAILABLE: "btn-ready",
    ACTIVE: "btn-green",
    CLEAN: "btn-light",
    REFERENCE: "btn-warning"
  };

  render() {
    let bankStyle = {};
    let cleanClassNames = "";
    let cleanDisabled = true;
    let getChannelClassNames = (_: number) => "";
    let getChannelDisabled = (_: number) => true;
    let test = {};

    if ("bank" in (this.props.uistatus as any)) {
      const bankStatus: string =
        this.props.uistatus.bank &&
        (this.props.uistatus.bank as any)[this.props.bank];
      const channelStatus: { [key: number]: string } =
        this.props.uistatus.channel &&
        (this.props.uistatus.channel as any)[this.props.bank];
      const cleanStatus: string =
        this.props.uistatus.clean &&
        (this.props.uistatus.clean as any)[this.props.bank];
      bankStyle = (this.bankStyleOpt as any)[bankStatus];
      cleanClassNames = (this.cleanClassNameOpt as any)[cleanStatus];
      cleanDisabled = cleanStatus !== "READY";
      getChannelClassNames = chan => {
        return (
          this.channelClassNameOpt !== undefined &&
          channelStatus !== undefined &&
          (this.channelClassNameOpt as any)[(channelStatus as any)[chan]]
        );
      };
      getChannelDisabled = chan => {
        return (
          channelStatus !== undefined &&
          (channelStatus as any)[chan] == "DISABLED"
        );
      };
      test = channelStatus;
    }

    const channelButtons = [];
    for (let i = 1; i <= 8; i++) {
      let portNumber = (this.props.bank - 1) * 8 + i
      channelButtons.push(
        getChannelDisabled(i) ? (
          <button
            key={portNumber}
            id={"channel-" + i}
            className={"btn btn-large bank-btn " + getChannelClassNames(i)}
            style={{ color: "black" }}
          >
            <p className="chn-label">
              <u className={"chn-name-" + i}>
              {portNumber + ": "}{this.props.plan.bank_names[this.props.bank].channels[i]}
              </u>
            </p>
            <p id={"chn-status-" + i} className={"chn-status"}>
              {" "}
            </p>
          </button>
        ) : (
          <button
            onClick={e => {
              this.props.addChanneltoPlan(this.props.bank, i);
            }}
            id={"channel-" + i}
            disabled={getChannelDisabled(i)}
            key={portNumber}
            className={"btn btn-large bank-btn " + getChannelClassNames(i)}
          >
            <p className="chn-label">
              <u className={"chn-name-" + i}>
                {portNumber + ": "}{this.props.plan.bank_names[this.props.bank].channels[i]}{" "}
              </u>
            </p>

            <p id={"chn-status-" + i} className={"chn-status"}>
              {" "}
            </p>
          </button>
        )
      );
    }

    const cleanButton = cleanDisabled ? (
      <div className={"btn btn-large btn-clean " + cleanClassNames}>
        {"Clean"}
      </div>
    ) : (
      <button
        id="clean"
        onClick={e =>
          // this.props.ws_sender({ element: "clean", bank: this.props.bank })
          this.props.addChanneltoPlan(this.props.bank, 0)
        }
        className={"btn btn-large btn-clean " + cleanClassNames}
      >
        {"Clean"}
      </button>
    );
    return (
      <div>
        <div className="panel-bank" style={bankStyle}>
          <div style={{ width: "100%" }}>
            <div className="grid-bank">{channelButtons}</div>
            {cleanButton}
          </div>
        </div>
      </div>
    );
  }
}
export default BankPanel;
