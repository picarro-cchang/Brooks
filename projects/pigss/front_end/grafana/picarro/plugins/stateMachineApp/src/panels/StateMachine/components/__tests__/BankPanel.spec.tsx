import React from 'react';
import BankPanel, {BankPanelOptions} from '../BankPanel';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';
import WS from "jest-websocket-mock";

const mockClick = jest.fn((element) => {return element;});
const apiLoc = `${window.location.hostname}:8000/controller`;
const socketURL = `ws://${apiLoc}/ws`;

const defaultProps: BankPanelOptions = {
    bank: 1,
    uistatus: {
      bank: { "1": "READY" },
      clean: { "1": "READY" },
      channel: { "1": 
      { 
          "1": "AVAILABLE",
          "2": "DISABLED",
          "3": "READY",
          "4": "DISABLED",
          "5": "AVAILABLE",
          "6": "DISABLED",
          "7": "AVAILABLE",
          "8": "DISABLED"
        },
            },
        },
        ws_sender: mockClick,
        plan: {
          bank_names: {
            1: {
              name: "B 1",
              channels: { 
                  1: "Ch A",
                  2: "Channel 2",
                  3: "Ch 3",
                  4: "Ch 4",
                  5: "Ch 5",
                  6: "Ch 6",
                  7: "Ch 7",
                  8: "Ch 8"
                 },
            },
          },
        }
    };



describe('<BankPanel />', () => {
  const part = mount(<BankPanel {...defaultProps} />);
  const wrapper = shallow(<BankPanel {...defaultProps} />);

  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('Contains Proper Channel Names', () => {
    const value1 = part.find('u.chn-name-1').text()
    const value2 = part.find('u.chn-name-2').text()
    expect(value1).toEqual('Ch A');
    expect(value2).toEqual('Channel 2');
  });

  it('Contains Proper Bank Name', () => {
    const value1 = part.find('h2').text()
    expect(value1).toEqual('B 1');
  });

  it('Contains correct status on channels', () => {
    const value = part.find('p#chn-status-1').text();
    expect(value).toEqual(' AVAILABLE ');
  });

  it('Check for  disabled/enabled property', () => {
    const value = part.find("button#channel-1").hasClass('disabled');
    const value2 = part.find("button#channel-2").hasClass('disabled');
    expect(value).toEqual(false);
    expect(value2).toEqual(true);
  });

  it("'WS onClick functionality'", async () => {
    const server = new WS(socketURL);
    const client = new WebSocket(socketURL);
    await server.connected;

    wrapper.find('button#channel-3').simulate('click');
    const element = mockClick.mock.calls[0][0];

    client.send(element);
    expect(mockClick).toHaveBeenCalled();
    await expect(server).toReceiveMessage({element: "channel", bank: 1, channel: 3});
    expect(mockClick).toHaveReturnedWith({bank: 1, channel: 3, element: "channel"})
    expect(server).toHaveReceivedMessages([{element: "channel", bank: 1, channel: 3}]);
  });
});

