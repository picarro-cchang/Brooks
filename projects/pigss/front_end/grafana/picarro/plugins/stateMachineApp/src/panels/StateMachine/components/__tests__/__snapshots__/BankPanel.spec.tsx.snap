// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BankPanel /> Renders correctly 1`] = `
<div>
  <div
    className="panel-bank"
    style={
      Object {
        "backgroundColor": "#56a64b",
        "color": "#fff",
      }
    }
  >
    <div
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <h2
        style={
          Object {
            "color": "#424242",
            "marginBottom": "10px",
          }
        }
      >
        B 1
      </h2>
      <div
        className="grid-bank"
      >
        <button
          className="btn btn-large bank-btn btn-ready"
          key="1"
          style={
            Object {
              "color": "black",
            }
          }
        >
          <p
            className="chn-label"
          >
            <u
              className="chn-name-1"
            >
              Ch A
            </u>
          </p>
          <p
            className="chn-status-1"
          >
             
            AVAILABLE
             
          </p>
        </button>
        <button
          className="btn btn-large bank-btn btn-inverse-2 disabled"
          key="2"
          style={
            Object {
              "color": "black",
            }
          }
        >
          <p
            className="chn-label"
          >
            <u
              className="chn-name-2"
            >
              Channel 2
            </u>
          </p>
          <p
            className="chn-status-2"
          >
             
            DISABLED
             
          </p>
        </button>
        <button
          className="btn btn-large bank-btn btn-ready"
          key="3"
          style={
            Object {
              "color": "black",
            }
          }
        >
          <p
            className="chn-label"
          >
            <u
              className="chn-name-3"
            >
              Ch 3
            </u>
          </p>
          <p
            className="chn-status-3"
          >
             
            AVAILABLE
             
          </p>
        </button>
        <button
          className="btn btn-large bank-btn btn-inverse-2 disabled"
          key="4"
          style={
            Object {
              "color": "black",
            }
          }
        >
          <p
            className="chn-label"
          >
            <u
              className="chn-name-4"
            >
              Ch 4
            </u>
          </p>
          <p
            className="chn-status-4"
          >
             
            DISABLED
             
          </p>
        </button>
        <button
          className="btn btn-large bank-btn btn-ready"
          key="5"
          style={
            Object {
              "color": "black",
            }
          }
        >
          <p
            className="chn-label"
          >
            <u
              className="chn-name-5"
            >
              Ch 5
            </u>
          </p>
          <p
            className="chn-status-5"
          >
             
            AVAILABLE
             
          </p>
        </button>
        <button
          className="btn btn-large bank-btn btn-inverse-2 disabled"
          key="6"
          style={
            Object {
              "color": "black",
            }
          }
        >
          <p
            className="chn-label"
          >
            <u
              className="chn-name-6"
            >
              Ch 6
            </u>
          </p>
          <p
            className="chn-status-6"
          >
             
            DISABLED
             
          </p>
        </button>
        <button
          className="btn btn-large bank-btn btn-ready"
          key="7"
          style={
            Object {
              "color": "black",
            }
          }
        >
          <p
            className="chn-label"
          >
            <u
              className="chn-name-7"
            >
              Ch 7
            </u>
          </p>
          <p
            className="chn-status-7"
          >
             
            AVAILABLE
             
          </p>
        </button>
        <button
          className="btn btn-large bank-btn btn-inverse-2 disabled"
          key="8"
          style={
            Object {
              "color": "black",
            }
          }
        >
          <p
            className="chn-label"
          >
            <u
              className="chn-name-8"
            >
              Ch 8
            </u>
          </p>
          <p
            className="chn-status-8"
          >
             
            DISABLED
             
          </p>
        </button>
      </div>
      <button
        className="btn btn-large btn-clean btn-ready"
        onClick={[Function]}
      >
        Clean
      </button>
    </div>
  </div>
</div>
`;
